core/primitives/tests/crate-limit-test.rs

```diff
@@ -0,0 +1,46 @@
+use std::collections::HashSet;
+use std::process::Command;
+use std::str;
+
+// when you compile you see line similar to the next one:
+// Building [=======        ] 50/100: near-primitives v0.1.0
+// The 100 is not actually the number of dependencies, but the number of crates that are being built.
+// This threshhold represents the number of dependencies
+const THRESHOLD: usize = 150;
+
+#[test]
+fn test_crate_count() {
+    // Run `cargo tree -p near-primitives --edges=normal` and capture the output
+    let output = Command::new("cargo")
+        .arg("tree")
+        .arg("-p")
+        .arg("near-primitives")
+        .arg("--edges=normal")
+        .output()
+        .expect("Failed to execute cargo tree");
+
+    // Check if the command was successful
+    assert!(output.status.success(), "Cargo tree failed");
+
+    // Convert the output to a string
+    let output_str = str::from_utf8(&output.stdout).expect("Failed to convert output to string");
+
+    let re = regex::Regex::new(r"([\w-]+) v([\d.]+(?:-\w+)?)").unwrap();
```

Comment:
This seems like its going one step forward two back. From what I could tell cargo outputs pretty much a line per package. Why not count those lines?

Reply:
(duplicated entries have `(*)` in their output, so you can do a `.contains()` check too)

Reply:
It works a bit differently... 
(*) shows that this dependency is already somewhere expanded so that it won't show sub-dependencies for that.

But the (*) wouldn't appear if there were no sub-dependencies. 

Here is a proof:
![image](https://github.com/near/nearcore/assets/42449190/6034bb93-54b2-46eb-9eac-49ba0cc92310)